source/fundamentals/auth.txt

```diff
@@ -229,145 +228,179 @@ connect using ``MONGODB-CR``.
 ``MONGODB-AWS``
 ~~~~~~~~~~~~~~~
 
+..
+   The MONGODB-AWS section structure was updated for v4.8 of the driver. Avoid
+   backporting any documentation that might not apply to a prior version.
+
 .. note::
 
-   The MONGODB-AWS authentication mechanism is available in MongoDB
-   Atlas.
+   The MONGODB-AWS authentication mechanism is available for MongoDB
+   deployments on MongoDB Atlas.
 
 The ``MONGODB-AWS`` authentication mechanism uses your Amazon Web Services
 Identity and Access Management (AWS IAM) credentials to authenticate your
-user.
+user. To learn more about configuring MongoDB Atlas, see the
+:atlas:`Set Up Passwordless Authentication with AWS IAM Roles </security/passwordless-authentication/#set-up-passwordless-authentication-with-aws-iam-roles>`
+guide.
 
-You can store your AWS credentials as environment variables, or insert
-them inline like the examples below. The driver checks for your credentials
-in the following order:
+You can authenticate using the ``MONGODB-AWS`` mechanism by providing
+your AWS IAM credentials in one of the following ways:
 
-1. Supplied values in a ``MongoCredential`` object or the provided connection string.
-2. Your environment variables. (``AWS_ACCESS_KEY_ID``, ``AWS_SECRET_ACCESS_KEY``,
-   and optionally ``AWS_SESSION_TOKEN``)
-3. The AWS EC2 endpoint specified in the ``AWS_CONTAINER_CREDENTIALS_RELATIVE_URI``
-   environment variable.
-4. The default AWS EC2 endpoint. For more information, see `IAM Roles for Tasks
-   <https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html>`__
+- As environment variables
```

Comment:
I'm still confused by this, based on the shared doc we are now working off of.

Reply:
I see, did I conflate "From ENVIRONMENT" with "environment variables"? If so, what else is from the environment that I missed that isn't the environment variables?
If not, doesn't the current organization correspond directly with "FROM ENVIRONMENT" and "FROM MongoClient CONFIGURATION"? There are only two headers within the section now which I intended to correspond to this bulleted list and the doc:
"Specify Your Credentials in Environment Variables"
"Specify Your Credentials in MongoClient Configuration"

Reply:
> Did I conflate "From ENVIRONMENT" with "environment variables"? If so, what else is from the environment that I missed that isn't the environment variables?

Yes, exactly.  The environment also includes the HTTP endpoints for EC2 and ECS

Reply:
Trying to figure out how to define "from environment" to a user -- what are the boundaries of that term? I can't think of any specific term that might express environment variables (for secret, tokens, ECS endpoint) and lack of environment variables (EC2 endpoint) which is what I think we intend to communicate.

Would "As environment variables or by using the AWS SDK for Java" be more appropriate?

Reply:
No, because the SDK also, and first, looks for the presence of env vars.